### DISCUSSION TITLE
Add a Testing Section

### DISCUSSION THREAD

**Discussion by councit** [markerikson/react-community-tools-practices-cheatsheet#24] [Ideas]
Looking at the site so far, I was wondering what the team thought about adding a Testing section. 

Topics covered to include: 
- Unit Testing
- Integration Tests
- E2E Testing

**Comment by srmagura** [comment]
Sounds good to me, though it could be good to get the stuff we've already scaffolded done before adding more areas to the site.

Some of my personal opinions on the different types of testing:

- I like Kent C. Dodds' testing trophy, https://kentcdodds.com/blog/the-testing-trophy-and-testing-classifications
- Unit testing tiny React components is not that impactful. Better to unit test complex logic like reducers.
- For E2E, I've used Playwright and Cypress. I believe Playwright is more powerful and intuitive, though Cypress still has more downloads on npm probably because it's been around longer.

**Reply by councit** [reply]
Great I will ready through this. I have used jest, jasmine, cypress, and karma. Totally agree to work through the sections before moving to this. Thanks for your response!

**Comment by nickserv** [comment]
For unit and integration tests, I’d recommend Vitest, but I want to improve the Testing Library docs around it first
